File: frontend/src/modules/country/api/countryApi.ts
import { axiosClient } from "../../../shared/api/axiosClient";

import type {
    CountrySummary,
    CountryResponse,
    CreateCountryRequest,
    UpdateCountryRequest,
} from "../types";

export const countryApi = {

    async getAll(): Promise<CountrySummary[]> {

        const response =
            await axiosClient.get<CountrySummary[]>("/countries");

        return response.data;

    },

    async create(
        request: CreateCountryRequest,
    ): Promise<CountryResponse> {

        const response =
            await axiosClient.post<CountryResponse>(
                "/countries",
                request,
            );

        return response.data;

    },

    async update(
        id: string,
        request: UpdateCountryRequest,
    ): Promise<CountryResponse> {

        const response =
            await axiosClient.put<CountryResponse>(
                `/countries/${id}`,
                request,
            );

        return response.data;

    },

    async delete(id: string): Promise<void> {

        await axiosClient.delete(`/countries/${id}`);

    },
    async activate(id: string): Promise<void> {

    await axiosClient.patch(
        `/countries/${id}/activate`,
    );

    },

async deactivate(id: string): Promise<void> {

    await axiosClient.patch(
        `/countries/${id}/deactivate`,
    );

    },
};

File: backend/src/main/java/org/afdb/aikp/modules/country/presentation/contract/CountryApi.java
package org.afdb.aikp.modules.country.presentation.contract;

import org.afdb.aikp.modules.country.application.response.CountryResponse;
import org.afdb.aikp.modules.country.application.response.CountrySummary;
import org.afdb.aikp.modules.country.presentation.request.CreateCountryRequest;
import org.afdb.aikp.modules.country.presentation.request.UpdateCountryRequest;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.UUID;

/**
 * Contract for Country REST operations.
 *
 * <p>
 * This interface defines the REST contract independently of the
 * Spring MVC/OpenAPI annotations, which are declared on the
 * implementing controller.
 * </p>
 */
public interface CountryApi {

    ResponseEntity<CountryResponse> create(
            CreateCountryRequest request);

    ResponseEntity<CountryResponse> get(
            UUID id);

    ResponseEntity<List<CountrySummary>> getAll();

    ResponseEntity<List<CountrySummary>> getActive();

    ResponseEntity<CountryResponse> update(
            UUID id,
            UpdateCountryRequest request);

    ResponseEntity<CountryResponse> activate(
            UUID id);

    ResponseEntity<CountryResponse> deactivate(
            UUID id);

    ResponseEntity<Void> delete(
            UUID id);

}